src/lib/JreSetupManager.ts

```diff
@@ -0,0 +1,183 @@
+import { CONFIG } from '../Constants';
+import { ConfigFile, Logger, SfdxError, Messages } from '@salesforce/core';
+import { AsyncCreatable } from '@salesforce/kit';
+
+import process = require('process');
+import findJavaHome = require('find-java-home');
+import childProcess = require('child_process');
+import path = require('path');
+import { FileHandler } from './FileHandler';
+
+// Initialize Messages with the current plugin directory
+Messages.importMessagesDirectory(__dirname);
+
+const JAVA_HOME_KEY = 'java-home';
+const JAVA_HOME_SYSTEM_VARIABLES = ['JAVA_HOME', 'JRE_HOME', 'JDK_HOME'];
+
+class Config extends ConfigFile<ConfigFile.Options> {
+
+    public static getFileName(): string {
+        // TODO: Revisit the file location!
+        // This doesn't work if I give the directory as ~/.sfdx-scanner.
+        // File only goes to ~/.sfdx
+        return CONFIG;
+    }
+}
+
+class JreSetupManager extends AsyncCreatable {
+    private logger!: Logger;
+    private config!: Config;
+
+    protected async init(): Promise<void> {
+        this.logger = await Logger.child('verifyJRE');
+        this.config = await Config.create({
+            isGlobal: true,
+            throwOnNotFound: false
+        });
+    }
+
+    async verifyJreSetup(): Promise<void> {
+        // Find Java Home
+        const javaHome = await this.findJavaHome();
+
+        // If we have javaHome, verify that it is a valid path
+        await this.verifyPath(javaHome);
+
+        // If path exists, verify Java version
+        await this.verifyJavaVersion(javaHome);
+
+        // Write javaHome to Config file
+        await this.setJavaHomeInConfig(javaHome);
+
+        return javaHome;
```

Comment:
Ongoing change to PmdWrapper has leaked here. Will be fixed when rest of the code is ready.